tests/mpi/petsc_step-27.cc

```diff
@@ -476,9 +472,10 @@ main(int argc, char *argv[])
       using namespace Step27;
 
       Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
-      // This is the right choice (e.g., vs. mpi_initlog()) since the test
-      // uses ConditionalOStream
-      initlog();
+
+      // We need mpi_initlog() in order to  make sure to only write to the
+      // »output« file from rank 0.
+      mpi_initlog();
```

Comment:
So here is the question regarding the "moreover" comment above. It's only one file. But only one process writes into it. Is the matter that multiple processes open it, and in the process truncate the file that other processes have already started to write into?

Reply:
The problem is the innocent looking `initlog()` which will `ofstream::open()` the same `output` file on two MPI ranks. `open()`  without extra argument implies that the file gets created if it does not already exist, and if it already exists truncated. There is a bit of ambiguity how this maps to system calls precisely but bottom line is that if rank 1 is late to the party then it will truncate the file after rank 0 has already written something to it.

Reply:
There is nothing wrong with the `pcout` above except that it wasn't necessary when using `mpi_initlog()` here.